docs/proposals/local-cluster-ux.md

```diff
@@ -0,0 +1,184 @@
+
+For running and managing the kubernetes components themselves,  we can re-use [Spread's localkube](https://github.com/redspread/localkube).
+Localkube is a self-contained go binary that includes all the master components including DNS and runs them using multiple go threads.
+Each Kubernetes release will include a localkube binary that has been tested exhaustively.
+
+To support Windows and OS X, nanokube will use [libmachine](https://github.com/docker/machine/tree/master/libmachine) internally to create and destroy virtual machines.
+Nanokube will be shipped with an hypervisor (virtualbox) in the case of OS X.
+Nanokube will include a base image that will be well tested.
+
+In the case of Linux, since the cluster can be run locally, we ideally want to avoid setting up a VM.
+Since docker is the only fully supported runtime as of Kubernetes v1.2, we can initially use docker to run and manage localkube.
+There is risk of being incompatible with the existing version of docker.
+By using a VM, we can avoid such incompatibility issues though.
+Feedback from the community will be helpful here.
+
+If the goal is to run outside of a VM, we can have nanokube prompt the user if docker is unavailable or version is incompatible.
+Alternatives to docker for running the localkube core includes using [rkt](https://coreos.com/rkt/docs/latest/), setting up systemd services, or a System V Init script depending on the distro.
+
+To summarize the pipeline is as follows:
+
+##### OS X / Windows
+
+nanokube -> libmachine -> virtualbox/hyper V -> linux VM -> localkube
+
+##### Linux
+
+nanokube -> docker -> localkube
+
+### Alternatives considered
+
+#### Bring your own docker
+
+##### Pros
+
+- Kubernetes users will probably already have it
+- No extra work for us
+- Only one VM/daemon, we can just reuse the existing one
+
+##### Cons
+
+- Not designed to be wrapped, may be unstable
+- Might make configuring networking difficult on OS X and Windows
+- Versioning and updates will be challenging. We can mitigate some of this with testing at HEAD, but we'll - inevitably hit situations where it's infeasible to work with multiple versions of docker.
+- There are lots of different ways to install docker, networking might be challenging if we try to support many paths.
+
+#### Vagrant
+
+##### Pros
+
+- We control the entire experience
+- Networking might be easier to build
+- Docker can't break us since we'll include a pinned version of Docker
+- Easier to support rkt or hyper in the future
+- Would let us run some things outside of containers (kubelet, maybe ingress/load balancers)
+
+##### Cons
+
+- More work
+- Extra resources (if the user is also running docker-machine)
+- Confusing if there are two docker daemons (images built in one can't be run in another)
+- Always needs a VM, even on Linux
+- Requires installing and possibly understanding Vagrant.
+
```

Comment:
Nope, virtualbox doesn't run on ARM (https://forums.virtualbox.org/viewtopic.php?f=9&t=65426)
If it would, nobody would use it anyway, 'cause it would be so slow.